File: webapi/Models/UserSettings.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace webapi.Models
{
    
    public class UserSettings
    {

        public int UserSettingsId {  get; set; }

        [Column("UserId")]
        [Required]
        public User User { get; set; } = null!;

        public string TableDensity { get; set; } = null!;

        public string AddInventoryDefaultGrade { get; set; } = null!;

        public string RemoveInventoryDefaultGrade { get; set; } = null!;

        public bool TableFullScreenByDefault { get; set; }

        public string AvatarColor { get; set; } = null!;

        public string AvatarText { get; set; } = null!;
    }
}

File: reactapp/src/models/UserSettings.ts
import { TableDensity } from "../enums/TableDensity.enum";

export interface UserSettings {
  tableDensity: TableDensity; // Default: "comfortable"
  addInventoryDefaultGrade: string; // Default: "B-FRN"
  removeInventoryDefaultGrade: string; // Default: "B-FRN"
  tableFullScreenByDefault: boolean; // Default: false
  avatarColor: string; // Default: ""
  avatarText: string; // Default: ""
}
